zaza/openstack/charm_tests/ovn/tests.py

```diff
@@ -701,3 +701,223 @@ def run_tests(self):
         self.run_package_change_test(
             'ovn-central',
             'ovn-central')
+
+
+class OVNCentralDownscaleTests(test_utils.BaseCharmTest):
+    """Tests for cluster-status and cluster-kick actions."""
+
+    SB_CMD = "ovs-appctl -t /var/run/ovn/ovnsb_db.ctl {}"
+    NB_CMD = "ovs-appctl -t /var/run/ovn/ovnnb_db.ctl {}"
+
+    def _cluster_status_action(self):
+        """Return Southbound and Northbound cluster status.
+
+        This function returns data as reported by "cluster-status" action
+        parsed into two dictionaries in the following order:
+        "Southbound status", "Northbound status"
+        """
+        yaml_load_err = "Status of '{}' could not be loaded as yaml:\n{}"
+        status_raw = zaza.model.run_action_on_leader("ovn-central",
+                                                     "cluster-status")
+        status_data = status_raw.data["results"]
+        # Verify expected items in the action result
+        self.assertIn("ovnnb", status_data)
+        self.assertIn("ovnsb", status_data)
+
+        try:
+            nb_status = yaml.safe_load(status_data["ovnnb"])
+        except yaml.YAMLError:
+            self.fail(yaml_load_err.format("northbound-cluster",
+                                           status_data["ovnnb"]))
+        try:
+            sb_status = yaml.safe_load(status_data["ovnsb"])
+        except yaml.YAMLError:
+            self.fail(yaml_load_err.format("southbound-cluster",
+                                           status_data["ovnsb"]))
+
+        return sb_status, nb_status
+
+    @staticmethod
+    def _add_unit(number_of_units=1):
+        """Add specified number of units to ovn-central application.
+
+        This function also waits unit the application reaches active state.
+        """
+        zaza.model.add_unit(
+            "ovn-central",
+            count=number_of_units,
+            wait_appear=True
+        )
+        zaza.model.wait_for_application_states()
```

Comment:
I took this for a spin and I see the test code moving forward before everything is ready when adding and removing units, leading to spurious failures.

Normally this is due to a race where the test code sees the target state before the charm has started doing anything, and the solution is often something like this:
https://github.com/openstack-charmers/zaza-openstack-tests/blob/827d81fc12201e525d6d57d2908ee9be71f02649/zaza/openstack/charm_tests/ovn/tests.py#L625-L627

However, that trick does not work here as zaza/libjuju appears to get itself into trouble when units (dis-)appear.

Adding a call to `block_untill_all_units_idle` before `wait_for_application_states` does appear to help, what do you think?:
```python
        zaza.model.add_unit(...)
        zaza.model.block_until_all_units_idle()
        zaza.model.wait_for_application_states()
```